### PR TITLE
MAINT: Updated polynomial to use fstrings

### PR DESCRIPTION
**Pull request by notanton**
In response to issue #15420 I have updated the files in `numpy/polynomial/` to use fstrings.

This is my first time contributing to the project, I have read the sections in the docs on contributing/workflow and as far as I can tell, everything is in order.

I ran the tests via `runtests.py` and all appear to pass. If anyone has any feedback, or there are any other changes required please let me know.

Anton

### CODE REVIEW

File: numpy/polynomial/_polybase.py [hunk 1]
```diff
@@ -336,20 +328,20 @@ def _repr_latex_(self):
...
-                coef_str = ' - {}'.format(self._repr_latex_scalar(-c))
+                coef_str = f" - {self._repr_latex_scalar(-c)}"
 
             # produce the string for the term
             term_str = self._repr_latex_term(i, term, needs_parens)
             if term_str == '1':
                 part = coef_str
             else:
-                part = r'{}\,{}'.format(coef_str, term_str)
+                part = f"{coef_str}\\,{term_str}"
```
Comment (eric-wieser) [inline numpy/polynomial/_polybase.py]:
```suggestion
                part = rf"{coef_str}\,{term_str}"
```

Comment (eric-wieser) [reply numpy/polynomial/_polybase.py]: Same throughout

File: numpy/polynomial/polyutils.py [hunk 2]
```diff
@@ -786,12 +786,11 @@ def _deprecate_as_int(x, desc):
         else:
             if ix == x:
                 warnings.warn(
-                    "In future, this will raise TypeError, as {} will need to "
-                    "be an integer not just an integral float."
-                    .format(desc),
+                    f"In future, this will raise TypeError, as {desc} will "
+                    "need tobe an integer not just an integral float.",
```
Comment (charris) [inline numpy/polynomial/polyutils.py]: Typo, "tobe" <- "to be"

Comment (charris) [reply numpy/polynomial/polyutils.py]: Always worth running a spell check when working with strings :)

Comment (notanton) [reply numpy/polynomial/polyutils.py]: Thank Charles, sorry about that one. Good catch.

File: numpy/polynomial/_polybase.py (hunk 1 above)
Comment (notanton) [reply numpy/polynomial/_polybase.py]: Thanks Eric, I've made those changes.

**Comment by charris** [comment]
Thanks @notanton .

**Pull request merged**